### DISCUSSION TITLE
ring-offset-color opacity will mix with ring-color, result is not as expected

### DISCUSSION THREAD

**Discussion by MikeLee0358** [tailwindlabs/tailwindcss#13375] [Help]
**What version of Tailwind CSS are you using?**

v3.4.1

**What build tool (or framework if it abstracts the build tool) are you using?**

tailwind play, code pen

**What version of Node.js are you using?**

none

**What browser are you using?**

Firefox

**What operating system are you using?**

macOS

**Reproduction URL**

tailwind play: https://play.tailwindcss.com/r0O6WPI4AA

code pen(expected like this): https://codepen.io/michaelee1/pen/abxNWRa?editors=1111

**Describe your issue**
In double rings (ring-red + ring-offset-blue) situation, setting ring-offset-blue  to be opacity will get purple(blue + red), not as I expected blue-opacity.

if I understand document v3.4.1 correctly, ring-offset-* is built on top of ring-*, if it's true, I got confused and want to know ideas behind:

// copy from tailwind play
.ring-8 {
  --tw-ring-offset-shadow: var(--tw-ring-inset) 0 0 0 var(--tw-ring-offset-width) var(--tw-ring-offset-color);
  --tw-ring-shadow: var(--tw-ring-inset) 0 0 0 calc(8px + var(--tw-ring-offset-width)) var(--tw-ring-color);
  box-shadow: var(--tw-ring-offset-shadow), var(--tw-ring-shadow), var(--tw-shadow, 0 0 #0000);
}

1. box-shadow: should be like 
   box-shadow: var(--tw-ring-shadow), var(--tw-ring-offset-shadow), var(--tw-shadow, 0 0 #0000);
    1 .ring-* 2. ring-offset-* and ring-* will override ring-offset-*, this will ensure color will not be mixed when set opacity
    
 2. I read box-shadow mdn according to syntax the forth is "spread-radius", I know tailwindcss implement  it with --tw-ring-offset,   this will lead confused with first & second  parameters "offset-x" "offset-y", personal suggestion: "offset" change to "spread" is more intuition.
 
it will be like this:

.ring-offset-8   -> .ring-spread-8
--tw-ring-offset-color -> --tw-ring-spread-color

if my word is hard to understand, check out code pen : - )

I am study in tailwind, and rewrite 3 times for practice, only ring/ring-offset let me confused when I read mdn, otherwise I learn a lot from tailwindcss, thank for you all, awesome job!

**Comment by adamwathan** [comment]
Hey! The reason this is happening is that these utilities are built with box shadows, and box shadows have no concept of an offset like an outline does. Spread just makes the box shadow thicker, and we are using that already.

The way the ring utilities work  is say you have ring-2 with ring-offset-4. What we actually do is make the regular ring 6px thick (2+4), and then cover part of it with the offset ring. So for it to actually give off the effect of being an offset, the ring offset color needs to be a solid color that matches the background of your site.

Unfortunately it's just not possible in CSS to set an offset on a box shadow.

Generally I would recommend using the `outline` utilities these days when doing this sort of thing, because in modern browsers the `outline` property finally follows the element's radius. This wasn't always true, which is why we created the ring utilities in the first place.

**Reply by MikeLee0358** [reply]
thanks! after your explanation all things make sense!
